src/main/java/org/openrewrite/java/migrate/apache/commons/lang/ApacheCommonsStringUtils.java

```diff
@@ -0,0 +1,607 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.apache.commons.lang;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+@SuppressWarnings("unused")
+public class ApacheCommonsStringUtils {
+
+    private static class Abbreviate {
+        @BeforeTemplate
+        String before(String s1, int width) {
+            return StringUtils.abbreviate(s1, width);
+        }
+
+        @AfterTemplate
+        String after(String s, int width) {
+            return s.substring(0, Math.min(s.length(), width));
+        }
+    }
+
+    private static class Capitalize {
+        @BeforeTemplate
+        String before(String s) {
+            return StringUtils.capitalize(s);
+        }
+
+        @AfterTemplate
+        String after(String s) {
+            return s.substring(0, 1).toUpperCase() + s.substring(1);
+        }
+    }
+
+    //TODO: The test for this recipe fails, I think it could be due to a `rewrite-templating` bug
+    private static class Chomp {
+        @BeforeTemplate
+        String before(String s) {
+            return StringUtils.chomp(s);
+        }
+
+        @AfterTemplate
+        String after(String s) {
+            return s.endsWith("\n") ? s.substring(0, s.length() - 1) : s;
+        }
+    }
+    private static class Chop {
+        @BeforeTemplate
+        String before(String s) {
+            return StringUtils.chop(s);
+        }
+
+        @AfterTemplate
+        String after(String s) {
+            return s.substring(0, s.length() - 1);
+        }
+    }
+
+    private static class Contains {
+        @BeforeTemplate
+        boolean before(String s, String search) {
+            return StringUtils.contains(s, search);
+        }
+
+        @AfterTemplate
+        boolean after(String s, String search) {
+            return s.contains(search);
+        }
+    }
+
+    private static class CountMatches {
```

Comment:
I think I would probably remove this rule. While it probably is possible to do something like this, we would have to properly escape the pattern (as the input to `countMatches()` isn't a regex) and also using regex has a much higher computational overhead.

We can however provide a rule for `countMatches(String, char)` by using something like `s.chars().filter(c -> c == pattern).count()` (or `s.codePoints().filter(c -> c == pattern).count()`).